### PR TITLE
feat: get encoded channel monitors

### PR DESCRIPTION
**Pull request by rolznz**
TODOs:
- [x] error handling - function should return an error instead of using empty arrays
- [x] review naming
- [ ] for some reason tests are failing

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1539,6 +1539,29 @@ impl Node {
 		self.payment_store.remove(&payment_id)
 	}
 
+	/// Alby: Return encoded channel monitors for a recovery of last resort
+	pub fn get_encoded_channel_monitors(&self) -> Result<Vec<KeyValue>, Error> {
+		let channel_monitor_store = Arc::clone(&self.kv_store);
```
Comment (rdmitr) [inline src/lib.rs:1544]: Nitpick: why clone these?..

**Review by rdmitr — APPROVED** [review]
Approving with one minor comment :)

File: src/lib.rs (hunk 1 above)
Comment (rolznz) [reply src/lib.rs:1549]: I am not sure why it is needed, but every other place it's done in this way, so am just following convention

**Comment by rolznz** [comment]
Added ability to restore monitors and force close channels

File: src/lib.rs (hunk 1 above)
Comment (rdmitr) [reply src/lib.rs:1549]: They do it when spawning tasks, which makes sense: the tasks receive the cloned handle ;)

**Pull request merged**